core/libs/shared/src/lazy.rs

```diff
@@ -138,13 +156,24 @@ impl<T: Stagable> LazyTree<T> {
         if let Some(name) = self.name.get(id) {
             return Ok(name.clone());
         }
-
-        let key = self.decrypt_key(id, account)?;
-        let name = self.find(id)?.secret_name().to_string(&key)?;
-        self.name.insert(*id, name.clone());
+        let id = if let Some(link) = self.link(id)? { link } else { *id };
```

Comment:
when you ask for the name of a file that's linked, you get the link's name instead (relied on by path service in addition to being desired behavior for clients)